File: Data Structures & Algorithms/is-anagram/submission-7.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {

        if (s.size() != t.size()) return false;
        
        int mp[26] = {};

        for (int i = 0; i < s.size(); ++i) mp[s[i] - 'a']++;

        for (int i = 0; i < t.size(); ++i) mp[t[i] - 'a']--;

        for (int i = 0; i < 26; ++i) {
            if (mp[i]) return false;
        }

        return true;

    }

};

File: Data Structures & Algorithms/validate-parentheses/submission-7.py
class Solution:
    def isValid(self, s: str) -> bool:
        st = []
        opened = ["[", "{", "("]
        closed = ["]", "}", ")"]
        
        for i in range (len(s)):
            char = s[i]
            if char in opened:
                st.append(char)
            else:
                if (len(st) == 0):
                    return False
                if closed.index(char) == opened.index(st[-1]):
                    st.pop()
                else:
                    return False
        return len(st) == 0


                